docs/HACKING.CRYPTO

```diff
@@ -362,19 +362,19 @@ Converts the absolute value of bn into big-endian form and store it at
 val. val must point to _libssh2_bn_bytes(bn) bytes of memory.
 Returns the length of the big-endian number.
 
-void _libssh2_bn_rand(_libssh2_bn *bn, int bits, int top, int bottom);
+int _libssh2_bn_rand(_libssh2_bn *bn, int bits, int top, int bottom);
 Generates a cryptographically strong pseudo-random number of bits in
 length and stores it in bn. If top is -1, the most significant bit of the
 random number can be zero. If top is 0, it is set to 1, and if top is 1, the
 two most significant bits of the number will be set to 1, so that the product
 of two such random numbers will always have 2*bits length. If bottom is true,
-the number will be odd.
+the number will be odd. Returns 1 on success, 0 otherwise.
```

Comment:
I just took a look at the OpenSSL and libgcrypt backends. OpenSSL uses 0 on success and -1 **or** 1 on error for non-static functions. I think this needs to be unified at some point. At least the libgcrypt backend is consistent in using 0 on success and -1 on error, just like WinCNG (probably because I based the WinCNG code on the libgcrypt backend).